library/src/main/java/com/leinardi/android/speeddial/SpeedDialView.java

```diff
@@ -805,8 +808,24 @@ private void toggle(boolean show, boolean animate) {
     }
 
     private void updateMainFabDrawable(boolean animate) {
+        boolean updated = false;
         if (isOpen()) {
             if (mMainFabOpenedDrawable != null) {
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && mMainFabOpenedDrawable instanceof AnimatedVectorDrawable) {
+                    updated = true;
```

Comment:
Could you please move

```
                    updated = true;
                    mMainFab.setImageDrawable(mMainFabOpenedDrawable);
```

outside the if/else here and on the else branch? If you move this code above the it, we could avoid some code duplication.

Something like this (I did not test it):
```java
                updated = true;
                mMainFab.setImageDrawable(mMainFabOpenedDrawable);
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N && mMainFabOpenedDrawable instanceof AnimatedVectorDrawable) {
                    ((AnimatedVectorDrawable) mMainFabOpenedDrawable).start();
                } else if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N && mMainFabOpenedDrawable instanceof AnimatedVectorDrawableCompat) {
                    ((AnimatedVectorDrawableCompat) mMainFabOpenedDrawable).start();
                } else if (mMainFabOpenedDrawable instanceof AnimationDrawable){
                    ((AnimationDrawable) mMainFabOpenedDrawable).start();
                }
```

Reply:
removed updated.

Reply:
Just curious: why did you remove `updated`? I just asked to move it outside the if, not to remove it completely. Was it not needed?

Anyway, there is still code duplication for `mMainFab.setImageDrawable(mMainFabOpenedDrawable);`: please move it outside the if/else.

Reply:
Yes, `updated` was not needed.
I changed the code. Check please.

Reply:
Thanks!